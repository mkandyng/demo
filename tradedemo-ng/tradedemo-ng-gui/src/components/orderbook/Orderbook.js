import React from "react";
import { connect } from "react-redux";
import { getOrderStatus } from "../../common/libs/orderbook";
import OrderbookView from "./OrderbookView";

/**
 * Component to display Orderbook
 */
function Orderbook(props) {

    const { orderbook } = props;

    const eventHandler = {
        getTheadThProps: () => {
              return {
                style: {
                     height: "30px"
                 }
              }
        },
        getTrProps: (state, rowInfo, column) => {
             const orderStatus = rowInfo?getOrderStatus(rowInfo.original.status):undefined
             const backgroundColor = orderStatus ? orderStatus.backgroundColor: "white";
             const color = orderStatus ? orderStatus.color: "black";
             return {
                  style: {
                      backgroundColor: backgroundColor,
                      color: color
                  }
             }
        }
    };

    return (
      <OrderbookView orderbook={orderbook}
                     eventHandler={eventHandler} />
    );
}

const mapStateToProps = state => ({ orderbook: state.orderbook });

// The HOC
export default connect(
    mapStateToProps
)(Orderbook);
