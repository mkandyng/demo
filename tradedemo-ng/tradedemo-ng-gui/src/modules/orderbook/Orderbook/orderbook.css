#bottomLayout .ReactTable {
  background: #fffdfd;
  font-size: 10px;
  text-align: center;
  height: 88%;
}

#bottomLayout .ReactTable .rt-th {
  font-weight: bold;
  background: #f7f7f7;
}

#bottomLayout .ReactTable .rt-tr-group:hover {
  background: #f7f7f7;
}

#bottomLayout .ReactTable .rt-td-group {
  background: #ece5e5;
}

#bottomLayout .ReactTable .instrument-column {
  text-align: left;
}
