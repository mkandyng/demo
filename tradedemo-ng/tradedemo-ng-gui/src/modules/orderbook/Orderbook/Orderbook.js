import React from "react";
import ReactTable from "react-table";
import { orderbookStatusEnum } from "../orderbookStatusEnum";
import PropTypes from 'prop-types';
import "./orderbook.css";

/**
 * Component to handle orderbook view and interaction
 */
export default function Orderbook(props) {

    const { orders } = props;

    const eventHandlers = {
        handleTableColumnHeader: handleTableColumnHeader,
        handleTableRow: handleTableRow
    }

    return (
        <OrderbookView orders={orders}
                   columns={orderBookHeaderColumns}
                   eventHandlers={eventHandlers} />
    );
}

export function OrderbookView({orders,
                               columns,
                               eventHandlers}) {
    return (
        <ReactTable
            defaultPageSize={10}
            showPageSizeOptions={false}
            data={orders}
            columns={columns}
            getTheadThProps={eventHandlers.handleTableColumnHeader}
            getTrProps={eventHandlers.handleTableRow}
        />
    );
}

export const orderBookHeaderColumns = [
    { Header: "Order Ref", accessor: "orderRef", width: 80 },
    { Header: "Symbol", accessor: "symbol", width: 50 },
    { Header: "Created", accessor: "created", width: 120 },
    { Header: "Last Updated", accessor: "lastUpdated", width: 120 },
    { Header: "Instrument", accessor: "instrument", className: "instrument-column" },
    { Header: "Status", accessor: "status", width: 70 },
    { Header: "Ccy", accessor: "ccy", width: 40 },
    { Header: "Buy/Sell", accessor: "buySell", width: 50 },
    { Header: "Qty", accessor: "quantity", width: 50 },
    { Header: "Executed", accessor: "executed", width: 60},
    { Header: "Order Type", accessor: "orderType", width: 70 },
    { Header: "Order Price", accessor: "price", width: 70 },
    { Header: "Price (avg)", accessor: "avgPrice", width: 70 },
    { Header: "Expiry Type", accessor: "expiryType", width: 70},
    { Header: "Expiry Date", accessor: "expiryDate", width: 70}
];

function handleTableColumnHeader() {
    return {
        style: {
            height: "30px"
        }
    }
}

function handleTableRow(state, rowInfo, column) {
     const orderStatus = rowInfo?orderbookStatusEnum.getOrderStatus(rowInfo.original.status):undefined
     const backgroundColor = orderStatus ? orderStatus.backgroundColor: "white";
     const color = orderStatus ? orderStatus.color: "black";
     return {
          style: {
              backgroundColor: backgroundColor,
              color: color
          }
     }
}

Orderbook.propTypes = {
    orders: PropTypes.arrayOf(PropTypes.object).isRequired
};

OrderbookView.propTypes = {
    orders: PropTypes.arrayOf(PropTypes.object).isRequired,
    columns: PropTypes.arrayOf(PropTypes.object).isRequired,
    eventHandlers: PropTypes.objectOf(PropTypes.func).isRequired
};
