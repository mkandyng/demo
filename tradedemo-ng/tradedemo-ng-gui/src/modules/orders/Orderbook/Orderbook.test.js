import React from "react";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";
import Orderbook from "./Orderbook";

/**
 * This is unit test of Orderbook component
 * It leverage snapshot to test the presentation
 */

describe("Orderbook", () => {
  const props = {
    orders: [
      {
        orderRef: "XA12345678",
        symbol: "AMZ"
      }
    ]
  };

  it("should render component comparing with previous snapshot shallow markup", () => {
    // Given

    // When
    const component = shallow(<Orderbook orders={props.orders}/>);

    // Then
    expect(toJson(component)).toMatchSnapshot();
  });

});
