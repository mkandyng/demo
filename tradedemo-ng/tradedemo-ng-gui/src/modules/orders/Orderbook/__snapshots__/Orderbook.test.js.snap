// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orderbook should render component comparing with previous snapshot shallow markup 1`] = `
<ReactTable
  AggregatedComponent={[Function]}
  ExpanderComponent={[Function]}
  FilterComponent={[Function]}
  LoadingComponent={[Function]}
  NoDataComponent={[Function]}
  PadRowComponent={[Function]}
  PaginationComponent={[Function]}
  PivotValueComponent={[Function]}
  ResizerComponent={[Function]}
  TableComponent={[Function]}
  TbodyComponent={[Function]}
  TdComponent={[Function]}
  TfootComponent={[Function]}
  ThComponent={[Function]}
  TheadComponent={[Function]}
  TrComponent={[Function]}
  TrGroupComponent={[Function]}
  aggregatedKey="_aggregated"
  className=""
  collapseOnDataChange={true}
  collapseOnPageChange={true}
  collapseOnSortingChange={true}
  column={
    Object {
      "Aggregated": undefined,
      "Cell": undefined,
      "Expander": undefined,
      "Filter": undefined,
      "Footer": undefined,
      "Header": undefined,
      "Pivot": undefined,
      "PivotValue": undefined,
      "Placeholder": undefined,
      "aggregate": undefined,
      "className": "",
      "filterAll": false,
      "filterMethod": undefined,
      "filterable": undefined,
      "footerClassName": "",
      "footerStyle": Object {},
      "getFooterProps": [Function],
      "getHeaderProps": [Function],
      "getProps": [Function],
      "headerClassName": "",
      "headerStyle": Object {},
      "minResizeWidth": 11,
      "minWidth": 100,
      "resizable": undefined,
      "show": true,
      "sortMethod": undefined,
      "sortable": undefined,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "Header": "Order Ref",
        "accessor": "orderRef",
        "width": 80,
      },
      Object {
        "Header": "Symbol",
        "accessor": "symbol",
        "width": 50,
      },
      Object {
        "Header": "Created",
        "accessor": "created",
        "width": 120,
      },
      Object {
        "Header": "Last Updated",
        "accessor": "lastUpdated",
        "width": 120,
      },
      Object {
        "Header": "Instrument",
        "accessor": "instrument",
        "className": "instrument-column",
      },
      Object {
        "Header": "Status",
        "accessor": "status",
        "width": 70,
      },
      Object {
        "Header": "Ccy",
        "accessor": "ccy",
        "width": 40,
      },
      Object {
        "Header": "Buy/Sell",
        "accessor": "buySell",
        "width": 50,
      },
      Object {
        "Header": "Qty",
        "accessor": "quantity",
        "width": 50,
      },
      Object {
        "Header": "Executed",
        "accessor": "executed",
        "width": 60,
      },
      Object {
        "Header": "Order Type",
        "accessor": "orderType",
        "width": 70,
      },
      Object {
        "Header": "Order Price",
        "accessor": "price",
        "width": 70,
      },
      Object {
        "Header": "Price (avg)",
        "accessor": "avgPrice",
        "width": 70,
      },
      Object {
        "Header": "Expiry Type",
        "accessor": "expiryType",
        "width": 70,
      },
      Object {
        "Header": "Expiry Date",
        "accessor": "expiryDate",
        "width": 70,
      },
    ]
  }
  data={
    Array [
      Object {
        "orderRef": "XA12345678",
        "symbol": "AMZ",
      },
    ]
  }
  defaultExpanded={Object {}}
  defaultFilterMethod={[Function]}
  defaultFiltered={Array []}
  defaultPage={0}
  defaultPageSize={10}
  defaultResized={Array []}
  defaultSortDesc={false}
  defaultSortMethod={[Function]}
  defaultSorted={Array []}
  expanderDefaults={
    Object {
      "filterable": false,
      "resizable": false,
      "sortable": false,
      "width": 35,
    }
  }
  filterable={false}
  freezeWhenExpanded={false}
  getLoadingProps={[Function]}
  getNoDataProps={[Function]}
  getPaginationProps={[Function]}
  getProps={[Function]}
  getResizerProps={[Function]}
  getTableProps={[Function]}
  getTbodyProps={[Function]}
  getTdProps={[Function]}
  getTfootProps={[Function]}
  getTfootTdProps={[Function]}
  getTfootTrProps={[Function]}
  getTheadFilterProps={[Function]}
  getTheadFilterThProps={[Function]}
  getTheadFilterTrProps={[Function]}
  getTheadGroupProps={[Function]}
  getTheadGroupThProps={[Function]}
  getTheadGroupTrProps={[Function]}
  getTheadProps={[Function]}
  getTheadThProps={[Function]}
  getTheadTrProps={[Function]}
  getTrGroupProps={[Function]}
  getTrProps={[Function]}
  groupedByPivotKey="_groupedByPivot"
  indexKey="_index"
  loading={false}
  loadingText="Loading..."
  multiSort={true}
  nestingLevelKey="_nestingLevel"
  nextText="Next"
  noDataText="No rows found"
  ofText="of"
  onFetchData={[Function]}
  originalKey="_original"
  pageJumpText="jump to page"
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
    ]
  }
  pageText="Page"
  pivotDefaults={Object {}}
  pivotIDKey="_pivotID"
  pivotValKey="_pivotVal"
  previousText="Previous"
  resizable={true}
  resolveData={[Function]}
  rowsSelectorText="rows per page"
  rowsText="rows"
  showPageJump={true}
  showPageSizeOptions={false}
  showPagination={true}
  showPaginationBottom={true}
  showPaginationTop={false}
  sortable={true}
  style={Object {}}
  subRowsKey="_subRows"
/>
`;
